### PR TITLE
175 enhance or add to code documentation classes functions etc

### PR DESCRIPTION
**Pull request by Ben-Rey**

### CODE REVIEW

**Comment by gregLibert** [comment]
@Ben-Rey same here, you should bring all the commit you did in parallel branches. Just focus on what is to be done in this issue and leave the rest for the other branches.

**Comment by gregLibert** [comment]
@Ben-Rey i reviewed two functions to let you have a better understanding of what I'm expecting from this king of issue.

Can you do the same things for the others ?

File: assembly/std/storage.ts [hunk 1]
```diff
@@ -239,14 +239,21 @@ export function get<T>(key: T): T {
...
 
 /**
- * Returns (key, value) in the datastore of the given address.
+ * Returns the value associated with the given `key` in the datastore of the contract at the specified `address`.
  *
- * @privateRemarks
- * TODO: explains what happens on missing key.
+ * @remarks
+ * If there is no value associated with the `key`, the function will throw an error.
+ * If the contract at the given address does not exist, the function will throw an error.
```
Comment (gregLibert) [inline assembly/std/storage.ts]: Those two remarks should be using @throws

File: assembly/std/storage.ts [hunk 2]
```diff
@@ -239,14 +239,21 @@ export function get<T>(key: T): T {
...
+ *
+ * @typeParam T - The type of the key-value pair. Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param address - The address of the contract whose datastore is being queried.
+ * @param key - The key whose associated value is to be retrieved from the datastore. It will be converted
```
Comment (gregLibert) [inline assembly/std/storage.ts:250]: Convert notion should be  in  aprivate remark just to be sure that the end user is not confused.

File: assembly/std/storage.ts (hunk 1 above)
Comment (gregLibert) [reply assembly/std/storage.ts]: We should also add in the module introduction a note explaining that thrown exception are uncachable and can be found in he event

File: assembly/std/storage.ts [hunk 3]
```diff
@@ -239,14 +239,21 @@ export function get<T>(key: T): T {
...
+ * to a `StaticArray<u8>` using the `toDatastoreFormat()` function.
+ *
+ * @returns The value associated with the given `key` in the datastore of the specified contract, or throws an
+ * error if no value is found.
```
Comment (gregLibert) [inline assembly/std/storage.ts:254]: Not specifically here, but the test and ERROR should be also added to this function

File: assembly/std/storage.ts [hunk 4]
```diff
@@ -263,13 +263,22 @@ export function getOf<T>(address: Address, key: T): T {
...
- * TODO: explains what happens on missing key.
- * TODO: explains security mechanisms
+ * @remarks
+ * If the key is not of type string, StaticArray<u8>, Args, or Uint8Array, an error will be thrown and
+ * the compilation will stop.
+ * If the caller is not authorized to delete the key-value pair, an error will be thrown.
+ * If the key does not exist an error will be thrown.
+ *
+ * @privateRemarks
+ * TODO: describe the security mechanisms involved in this operation.
```
Comment (gregLibert) [inline assembly/std/storage.ts:276]: Should be in @remarks and should have an issue for the run team to come and explain this.

File: assembly/std/storage.ts [hunk 5]
```diff
@@ -263,13 +263,22 @@ export function getOf<T>(address: Address, key: T): T {
...
+ *
+ * @typeParam T - The type of the key to delete. Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param key - The key to delete from the datastore. It will be converted to a `StaticArray<u8>` using
```
Comment (gregLibert) [inline assembly/std/storage.ts:280]: Same as previous: imlementation (using toDatastoreFroamt) shoud be a privateRemarks.

File: assembly/std/storage.ts [hunk 6]
```diff
@@ -263,13 +263,22 @@ export function getOf<T>(address: Address, key: T): T {
...
+ * the `toDatastoreFormat()` function.
```
Comment (gregLibert) [inline assembly/std/storage.ts:281]: We should also call ERROR as soon as possible.

File: assembly/std/storage.ts (hunk 4 above)
Comment (Ben-Rey) [reply assembly/std/storage.ts:276]: Should I make one issue for each functions that need security mechanisms description. I already made a general one but I can transform it to meta issue

Comment (gregLibert) [reply assembly/std/storage.ts:276]: Both are ok. But you should be explicite (list precisely all that need to be done)

**Comment by Ben-Rey** [comment]
@gregLibert can we close this one ?

**Comment by gregLibert** [comment]
> @gregLibert can we close this one ?

Did you extract everything produce here to other issues/PR, if yes  then you can cloise it

**Pull request closed without merging**